cranelift/interpreter/src/value.rs

```diff
@@ -294,8 +294,7 @@ impl Value for DataValue {
         Ok(match kind {
             ValueConversionKind::Exact(ty) => match (self, ty) {
                 // TODO a lot to do here: from bmask to ireduce to raw_bitcast...
-                (DataValue::I64(n), types::I32) => DataValue::I32(i32::try_from(n)?),
-                (DataValue::I64(n), types::I64) => DataValue::I64(n),
+                (DataValue::I64(n), ty) if ty.is_int() => DataValue::from_integer(n as i128, ty)?,
                 (DataValue::I64(n), types::I128) => DataValue::I128(n as i128),
```

Comment:
Maybe I'm reading this wrong, but if `ty.is_int()` matches then the `types::I128` case is unreachable, right? Seems like that case should be deleted too.

Reply:
That's true, I missed that. Thanks!